src/Simple.OData.Client.V3.Adapter/RequestWriter.cs

```diff
@@ -18,17 +18,44 @@ namespace Simple.OData.Client.V3.Adapter
     public class RequestWriter : RequestWriterBase
     {
         private readonly IEdmModel _model;
+        
+        private Func<IODataRequestMessage, ODataMessageWriterSettings, IEdmModel, ODataMessageWriter> _messageWriterInitializer;
```

Comment:
This should be `readonly`.
```suggestion
        private readonly Func<IODataRequestMessage, ODataMessageWriterSettings, IEdmModel, ODataMessageWriter> _messageWriterInitializer;
```
This also looks like a good candidate for an interface, but that's up for debate.